### PR TITLE
Shot class

### PR DESCRIPTION
**Pull request by ivan-BRHM**
Created the universal shot class. The program works fine. The features were sort of intertwined so it was hard to split into fewer commits. But hopefully we can merge it without too much trouble.

### CODE REVIEW

**Comment by ivan-BRHM** [comment]
oh yea - the quick restart commit spilled over from a rebase. Had some shelved changes on another branch - dont mind the commit its removed again in the latter commit.

File: my_game.py [hunk 1]
```diff
@@ -45,6 +45,54 @@ def wrap(sprite: arcade.Sprite):
...
+        """
+        move the sprite and fade
+        """
+
+        self.center_x += self.change_x
+        self.center_y += self.change_y
+
+        wrap(self)
+
+        # check if the shot traveled too far
```
Comment (tobiasbp) [inline my_game.py:85]: This is not a check

File: my_game.py [hunk 2]
```diff
@@ -45,6 +45,54 @@ def wrap(sprite: arcade.Sprite):
...
+        self.range = range
+        self.distance_traveled = 0
+
+        self.forward(self.speed)
+
+        # play the shot sound if present
+        if sound:
+            sound.play()
+
+    def update(self):
```
Comment (tobiasbp) [inline my_game.py:75]: We should use _on_update()_ with _delta_time_

**Review by tobiasbp — CHANGES_REQUESTED** [review]
Minor changes needed

File: my_game.py (hunk 1 above)
Comment (ivan-BRHM) [reply my_game.py:85]: well, not the line - but the comment was meant to  account for both adding to distance traveled, and checking if it exceeds range

File: my_game.py (hunk 2 above)
Comment (ivan-BRHM) [reply my_game.py:75]: well, it isnt so in our other obj-classes... except for player - so we should shift it everywhere?

File: my_game.py (hunk 1 above)
Comment (ivan-BRHM) [reply my_game.py:85]: also... we dont need delta time - so why take unnecessary arguments?
Arcade uses update too.

File: my_game.py (hunk 2 above)
Comment (tobiasbp) [reply my_game.py:75]: Yes, we should use _on_update()_ everywhere.

**Review by tobiasbp — APPROVED** [review]

**Pull request merged**